Content.Shared/Disposal/Components/SharedDisposalUnitComponent.cs

```diff
@@ -17,6 +18,18 @@ public abstract partial class SharedDisposalUnitComponent : Component
     [ViewVariables(VVAccess.ReadWrite), DataField("soundFlush")]
     public SoundSpecifier? FlushSound = new SoundPathSpecifier("/Audio/Machines/disposalflush.ogg");
 
+    /// <summary>
+    /// Blacklists (prevents) entities listed from being placed inside.
+    /// </summary>
+    [DataField]
+    public EntityWhitelist? Blacklist;
+
+    /// <summary>
+    /// Whitelists (allows) entities listed from being placed inside.
+    /// </summary>
+    [DataField]
+    public EntityWhitelist? Whitelist;
+
```

Comment:
"MobsCanEnter" needs updating to account for this.

Reply:
I think mobscanenter works differently to a whitelist because it checks for mobstate alive. if dead a mob that cannot enter, can enter i believe.

Reply:
removed mobscanenter, this can be handled with either blacklist or whitelist like you requested.